npg_qc_viewer/lib/npg_qc_viewer/Controller/QcOutcomes.pm

```diff
@@ -158,12 +158,42 @@ sub _update_outcomes {
 
     try {
       my $seq_outcomes = $data->{$SEQ_OUTCOMES} || {};
-      my $lane_info = keys %{$seq_outcomes} ?
-        $self->_lane_info($c->model('MLWarehouseDB'), $seq_outcomes) : {};
+      my @lane_rps = keys %{$seq_outcomes};
+      my $lane_info = {};
+      # Why do we have to know the manufacturer?
+      # We are pushing Element Biosciences pooled libraries through this system.
+      # For time being (May 2025) no library-level autoqc data is available.
+      # The lanes' QC outcomes has to be finalised in the absenve of plex-level
+      # QC outcomes. 
+      my $illumina_instr_manufacturer = 1;
+      if (@lane_rps) {
+         my @inflated_lane_rps =
+           map { npg_tracking::glossary::rpt->inflate_rpt($_) }
+           @lane_rps;
+         my @id_runs = uniq map { $_->{'id_run'} } @inflated_lane_rps;
+         if (@id_runs == 1) { # In the SeqQC viewer all data will belong to
+                              # the same run.
+           my $id_run = $id_runs[0];
+           my $tracking_run = $c->model('NpgDB')->schema()->resultset('Run')
+                              ->find($id_run);
+           if (!$tracking_run) {
+             $self->status_not_found(
+               $c,
+               message => "Run $id_run is not registered in the tracking database"
+             );
+             return;
+           }
+           $illumina_instr_manufacturer = $tracking_run->instrument()
+                                          ->manufacturer_is_Illumina();
+         }
+         if ($illumina_instr_manufacturer) {
+           $lane_info = $self->_lane_info($c->model('MLWarehouseDB'), \@lane_rps);
+         }
+      }
 
       my $outcomes = npg_qc::mqc::outcomes->new(
-                       qc_schema => $c->model('NpgQcDB')->schema())
-                     ->save($data, $username, $lane_info);
+        qc_schema => $c->model('NpgQcDB')->schema()
+      )->save($data, $username, $lane_info, !$illumina_instr_manufacturer);
```

Comment:
True or false: We can save a QC outcome with no lane info if we derive 0 or 2+ id_runs from the composition OR it's an alternate manufacturer.

There's a lot of degrees of freedom here, and I'm not sure how many are constrained by context. Is a lane info not desirable for an elembio run?

Reply:
Lane info is information about tags in a pool. At the point of finalising QC outcome for a lane prelim. QC outcomes for all samples of a pool should be available for all tags. If available, they are finalised as well. Any mismatch or absence of info causes an error.

For Elembio we were asked to enable QC for lanes with multiple samples on lane level. That's why this relaxation is needed.

The UI is written in such a way that manual QC functionality is enabled for a single run only. Therefore I did not try to put any restrictions for multiple runs, a scenario does exist in practice.